keps/sig-apps/3088-pdb-for-daemonset-pods/README.md

```diff
@@ -0,0 +1,265 @@
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Support podDusruptionBudget (PDB) for DaemonSet pods. DaemonSet pods are evicted all at once
+ignoring PDB. It creates a load spike for kube-apiserver especially in relatively big clusters. This
+KEP proposes to change this behavior so that users could rely on PDB for DaemonSet pods like for all
+other built-in controllers.
+
+## Motivation
+
+PodDisruptionBudget can be not only a tool to keep user apps available during evictions, but also a
+mean of reducing of load on kube-apiserver. The can be caused by a restart of big number of
+DaemonSet pods due to eviction.
+
+Additionally, the disruption controller tracks all built-in pod controllers except DaemonSets when
+calculates allowed disruption. This behavior in inconsistent across built-in pod controllers and may
+cause confusion.
+
+### Goals
+
+- Support the disruption budget for DaemonSet the same way it works for other built-in pod
+  controllers.
+- Reduce the load to kube-apiserver caused by the recreation of DaemonSet pods being evicted.
+
+### Non-Goals
+
+- Introduce API changes.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+For now, PDB does not let to control DaemonSet pods evictions. If one creates PDB for DaemonSet
```

Comment:
This sounds like a bug, and moreover, if arbitrary controlled pods can't be used with a PDB, that's also a bug that should be fixed.

Reply:
I would expect any controller to work with PDBs  as described in the docs

> only .spec.minAvailable can be used, not .spec.maxUnavailable.

https://kubernetes.io/docs/tasks/run-application/configure-pdb/#arbitrary-controllers-and-selectors, if that's regressed we should know why.